addons/mail_plugin/controllers/mail_plugin.py

```diff
@@ -366,6 +370,18 @@ def _get_partner_data(self, partner):
         partner_values['title'] = partner.function
         partner_values['enrichment_info'] = None
 
+        if not partner_values['name']:
+            # Always ensure that the partner has a name
+            partner_values['name'] = partner_values['email']
```

Comment:
Maybe we could use _parse_partner_name on email and try to find a name and an email ?